stanford_profile.post_update.php

```diff
@@ -46,3 +49,56 @@ function stanford_profile_post_update_8014() {
   ]);
 
 }
+
+/**
+ * Create blocks for sites with custom themes that were added to stanford_basic.
+ */
+function stanford_profile_post_update_8015() {
+  $theme_name = \Drupal::config('system.theme')->get('default');
+  // Default theme is good. Just end if so.
+  if ($theme_name == "stanford_basic") {
+    return;
+  }
+
+  // Not stanford_basic, we have to create two config page blocks.
+  // Copy the blocks from stanford_basic and rename them.
+  //
+  // Names of things.
+  $basic_global_name = 'block.block.stanford_basic_config_pages_stanford_global_msg';
+  $basic_super_name = 'block.block.stanford_basic_config_pages_stanford_super_footer';
+  $my_global_name = 'block.block.' . $theme_name . '_config_pages_stanford_global_msg';
+  $my_super_name = 'block.block.' . $theme_name . '_config_pages_stanford_super_footer';
+
+  // Resources.
+  $config_path = Settings::get('config_sync_directory');
+  $source = new FileStorage($config_path);
+  $config_storage = \Drupal::service('config.storage');
+  $config_factory = \Drupal::service('config.factory');
+
+  // Get the configuration out of the filesystem as it may not have been imported
+  // yet...
+  $basic_global_config = $config_factory
+    ->getEditable($my_global_name)
+    ->setData(
+      $source->read($basic_global_name)
+    );
+  $basic_super_config = $config_factory
+    ->getEditable($my_super_name)
+    ->setData(
+      $source->read($basic_super_name)
+    );
+
+  // Change a few ids.
+  $basic_global_config->set('id', $theme_name . '_config_pages_stanford_global_msg');
+  $basic_super_config->set('id', $theme_name . '_config_pages_super_global_msg');
+  $basic_global_config->set('theme', $theme_name);
+  $basic_super_config->set('theme', $theme_name);
```

Comment:
dont forget to change the dependency in the config.

Reply:
Good catch. Thanks.

Reply:
Done.